### PR TITLE
Commit generated files automatically

### PR DESCRIPTION
**Pull request by GromNaN**
Instead of asking us to regenerate files locally when forgotten, it might as well be done directly by the GitHub Action.

Example of commit: https://github.com/mongodb/specifications/commit/7f27d57440da24c28afadea6b143993b17d692f0

### CODE REVIEW

**Review by alcaeus — CHANGES_REQUESTED** [review]

File: .github/workflows/json-regenerate-check.yml [hunk 1]
```diff
@@ -31,4 +31,9 @@ jobs:
           python3 ./source/client-side-encryption/etc/generate-test.py ./source/client-side-encryption/etc/test-templates/*.template ./source/client-side-encryption/tests/legacy
           python3 ./source/client-side-operations-timeout/etc/generate-basic-tests.py ./source/client-side-operations-timeout/etc/templates ./source/client-side-operations-timeout/tests
           python3 ./source/etc/generate-handshakeError-tests.py
-          cd source && make -B && git diff --exit-code
+          cd source && make -B
+
+      - name: "Commit the changes"
+        uses: stefanzweifel/git-auto-commit-action@v5
```
[the remark this answers is not in the document]
Comment (GromNaN) [reply .github/workflows/json-regenerate-check.yml:39]: Added, but that was not necessary when I tried [on my own fork](https://github.com/GromNaN/mongodb-specifications/pull/1).

[the remark this answers is not in the document]
Comment (GromNaN) [reply .github/workflows/json-regenerate-check.yml:39]: Yes, that leads to this commit: https://github.com/mongodb/specifications/commit/7f27d57440da24c28afadea6b143993b17d692f0

Let me try again.

[the remark this answers is not in the document]
Comment (GromNaN) [reply .github/workflows/json-regenerate-check.yml:39]: image

**Review by alcaeus — APPROVED** [review]

**Comment by alcaeus** [comment]
@GromNaN I believe rebasing this branch on top of master will fix the lint failure (which was fixed by #1689).
